File: systray_linux.c
#include <stdlib.h>
#include <stdio.h>
#include "systray.h"

void registerSystray(void) {
}

int nativeLoop(void) {
	printf("Linux is currently unsupported, failing to show system tray icon\n");
	return 0;
}

void setIcon(const char* iconBytes, int length, bool template) {
}

void setTitle(char* ctitle) {
}

void setTooltip(char* ctooltip) {
}

void setMenuItemIcon(const char* iconBytes, int length, int menuId, bool template) {
}

void add_or_update_menu_item(int menu_id, int parent_menu_id, char* title, char* tooltip, short disabled, short checked, short isCheckable) {
}

void add_separator(int menu_id) {
}

void hide_menu_item(int menu_id) {
}

void show_menu_item(int menu_id) {
}

void quit() {
}

File: systray_linux.go
package systray

// SetTemplateIcon sets the systray icon as a template icon (on macOS), falling back
// to a regular icon on other platforms.
// templateIconBytes and iconBytes should be the content of .ico for windows and
// .ico/.jpg/.png for other platforms.
func SetTemplateIcon(templateIconBytes []byte, regularIconBytes []byte) {
}


// SetIcon sets the systray icon.
// iconBytes should be the content of .ico for windows and .ico/.jpg/.png
// for other platforms.
func SetIcon(iconBytes []byte) {
}

// SetTitle sets the systray title, only available on Mac and Linux.
func SetTitle(title string) {
}

// SetTooltip sets the systray tooltip to display on mouse hover of the tray icon,
// only available on Mac and Windows.
func SetTooltip(tooltip string) {
}

// SetIcon sets the icon of a menu item. Only works on macOS and Windows.
// iconBytes should be the content of .ico/.jpg/.png
func (item *MenuItem) SetIcon(iconBytes []byte) {
}

// SetTemplateIcon sets the icon of a menu item as a template icon (on macOS). On Windows, it
// falls back to the regular icon bytes and on Linux it does nothing.
// templateIconBytes and regularIconBytes should be the content of .ico for windows and
// .ico/.jpg/.png for other platforms.
func (item *MenuItem) SetTemplateIcon(templateIconBytes []byte, regularIconBytes []byte) {
}

func addOrUpdateMenuItem(item *MenuItem) {
}

func setInternalLoop(internal bool) {
}

func registerSystray() {
}

func addSeparator(id uint32) {
}

func hideMenuItem(item *MenuItem) {
}

func showMenuItem(item *MenuItem) {
}
func nativeLoop() int {
	return 0
}

func nativeEnd() {
	systrayExit()
}

func quit() {
}

func nativeStart() {
	systrayReady()
}
